WordPress/Classes/Networking/Pinghub.swift

```diff
@@ -0,0 +1,191 @@
+import Foundation
+import Starscream
+
+
+// MARK: Client
+
+
+/// The delegate of a PinghubClient must adopt the PinghubClientDelegate
+/// protocol. The client will inform the delegate of any relevant events.
+///
+public protocol PinghubClientDelegate {
+    /// The client connected successfully.
+    ///
+    func pingubConnected(client: PinghubClient)
+
+    /// The client disconnected. This might be intentional or due to an error.
+    /// The optional error argument will contain the error if there is one.
+    ///
+    func pinghubDisconnected(client: PinghubClient, error: Error?)
+
+    /// The client received an action.
+    ///
+    func pinghubActionReceived(client: PinghubClient, action: PinghubClient.Action)
+
+    /// The client received some data that it didn't look like a known action.
+    ///
+    func pinghubUnexpectedDataReceived(client: PinghubClient, message: String)
```

Comment:
(Suggestion). I don't know if this is a rule, but i'm really used to seeing methods that look like this, perhaps.


```
func pinghubDidConnect(_ pinghub: PinghubClient)
func pinghubDidDisconnect(_ pinghub: PinghubClient)
func pinghub(_ pinghub: PinghubClient, didReceiveAction action: PinghubClient.Action)
func pinghub(_ pinghub: PinghubClient, didReceiveMessage message: String)
```